src/main/java/org/runejs/client/LinkedList.java

```diff
@@ -66,113 +66,113 @@ else if(Buffer.anInt1985 != Player.worldLevel) {
     }
 
     public void pushBack(Node node, int arg1) {
         if(node.previous != null)
             node.remove();
-        node.previous = aClass40_1056.previous;
-        int i = 106 % ((-7 - arg1) / 59);
-        node.next = aClass40_1056;
+
+        node.previous = tail.previous;
+        node.next = tail;
         node.previous.next = node;
         node.next.previous = node;
     }
 
-    public void method905(int arg0, Node arg1) {
-        if(arg1.previous != null)
-            arg1.remove();
-        if(arg0 != 0)
-            method911(-96, null, null);
-        arg1.next = aClass40_1056.next;
-        arg1.previous = aClass40_1056;
-        arg1.previous.next = arg1;
-        arg1.next.previous = arg1;
+    public void pushFront(int arg0, Node node) {
+        if(node.previous != null)
+            node.remove();
+
+        node.previous = tail;
+        node.next = tail.next;
+        node.previous.next = node;
+        node.next.previous = node;
     }
 
     public void clear(int arg0) {
-        if(arg0 != 0)
-            method903(-33, (byte) -75);
-        for(; ; ) {
-            Node class40 = aClass40_1056.next;
-            if(class40 == aClass40_1056)
-                break;
-            class40.remove();
+        while (true) {
+            Node next = tail.next;
+
+            if(next == tail)
+                return;
+
+            next.remove();
         }
     }
 
-    public Node method909(int arg0) {
-        Node class40 = aClass40_1068;
-        if(aClass40_1056 == class40) {
-            aClass40_1068 = null;
+    public Node next(int arg0) {
+        Node node = current;
+
+        if(node == tail) {
+            current = null;
             return null;
         }
-        if(arg0 != -4)
-            method910(109);
-        aClass40_1068 = class40.next;
-        return class40;
+
+        current = node.next;
+        return node;
     }
 
     public void method911(int arg0, Node arg1, Node arg2) {
         if(arg2.previous != null)
             arg2.remove();
         if(arg0 == -31793) {
-            arg2.next = arg1;
             arg2.previous = arg1.previous;
+            arg2.next = arg1;
             arg2.previous.next = arg2;
             arg2.next.previous = arg2;
         }
     }
 
     public Node previous(int arg0) {
-        if(arg0 != 4)
-            return null;
-        Node class40 = aClass40_1068;
-        if(class40 == aClass40_1056) {
-            aClass40_1068 = null;
+        Node node = current;
+
+        if(node == tail) {
+            current = null;
             return null;
         }
-        aClass40_1068 = class40.previous;
-        return class40;
+
+        current = node.previous;
+        return node;
     }
 
-    public Node method913(int arg0) {
-        if(arg0 != 25447)
-            method902((byte) -115);
-        Node class40 = aClass40_1056.next;
-        if(aClass40_1056 == class40)
+    public Node popFront(int arg0) {
+        Node next = tail.next;
```

Comment:
arg0 != 25447 was removed what was it doing?

Reply:
just a dummy variable added by the obfuscator, was effectively unused (all code paths passed it in as `25447`)